sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -66,17 +65,16 @@ static bool _dyn_port_used(uint16_t port)
 /**
  * @brief   returns a UDP port, and checks for reuse if required
  *
- * complies to RFC 6056, see https://tools.ietf.org/html/rfc6056#section-3.3.3
+ * implements "Another Simple Port Randomization Algorithm" as specified in
+ * RFC 6056, see https://tools.ietf.org/html/rfc6056#section-3.3.2
  */
 static uint16_t _get_dyn_port(sock_udp_t *sock)
 {
     unsigned count = GNRC_SOCK_DYN_PORTRANGE_NUM;
     do {
         uint16_t port = GNRC_SOCK_DYN_PORTRANGE_MIN +
-               (_dyn_port_next * GNRC_SOCK_DYN_PORTRANGE_OFF) % GNRC_SOCK_DYN_PORTRANGE_NUM;
-        _dyn_port_next++;
-        if ((sock == NULL) || (sock->flags & SOCK_FLAGS_REUSE_EP) ||
-                              !_dyn_port_used(port)) {
+               (random_uint32() % GNRC_SOCK_DYN_PORTRANGE_NUM);
+        if ((sock && (sock->flags & SOCK_FLAGS_REUSE_EP)) || !_dyn_port_used(port)) {
```

Comment:
`&&`?  I'd suggest to leave as it was before, right?

Reply:
I thought [we agreed](https://github.com/RIOT-OS/RIOT/pull/13253#discussion_r375118405) that the previous check was broken:

> mhm, you're right the '||' doesn't make sense here.

The previous check would not check for local port collisions if `sock` is NULL, which seems very wrong to me.

Reply:
?